### ISSUE TITLE
75f13c9eea changed the function prototype for Mage_GiftMessage_Model_Observer->checkoutEventCreateGiftMessage breaking Ebizmarts Sagepaysuite

### ISSUE THREAD

**Issue by rbrown** [OpenMage/magento-lts#1208]
Hi, 75f13c9eea includes the following changes to the method `Mage_GiftMessage_Model_Observer->checkoutEventCreateGiftMessage`

```diff
      * Operate with gift messages on checkout proccess
      *
-     * @param Varieb_Object $observer
+     * @param c $observer
      * @return $this
      */
-    public function checkoutEventCreateGiftMessage($observer)
+    public function checkoutEventCreateGiftMessage(Varien_Event_Observer $observer)
     {
         $giftMessages = $observer->getEvent()->getRequest()->getParam('giftmessage');
```

Ebizmarts Sagepaysuitepro includes code similar to

```php
        $event = new Varien_Object;
        Mage::getModel('giftmessage/observer')->checkoutEventCreateGiftMessage($event);
```

This was the correct object to create according to the docblock, but now causes php to die because Varien_Object is not a child of Varien_Event_Observer.

I guess my questions are:
1. Should the prototype have been changed in 19.4.x
2. If it should, should I be contacting Ebizmarts  to request they update their code to use. a Varien_Event_Observer?

**Comment by colinmollenhour** [comment]
I've seen code like this before as well.  A good compromise would be to use `Varien_Object` in the type declaration since `Varien_Event_Observer` extends `Varien_Object`.

**Comment by sreichel** [comment]
Sorry for introducing this ... 

It may be a BC breaking change, but i would prefer a change in 3rd party code an keep/make core code consistent.

**Comment by rbrown** [comment]
I'm a big fan of consistency, but with strict typing this change is breaking an interface that, according to the history in this repo, was created in 2009 in v1.1.1 d19f0e13d 

When this interface is broken like this it stops customers being able to place orders using this module

**Comment by fballiano** [comment]
Hi everybody, just to check, is this issue still valid? should we do something about it?

**Comment by rbrown** [comment]
Yes, there's been no change in the downstream module, and openmage should (I think)  change the type declaration to Varien_Object as Colin says in https://github.com/OpenMage/magento-lts/issues/1208#issuecomment-693113573 to preserve the historical contract.

**Comment by fballiano** [comment]
I've created a PR for this, let's see if people will approve it

**Comment by fballiano** [comment]
Since it's now 3 years it doesn't make sense to me to implement a slightly wrong interface for this. We shouldn't have changed the "interface" in the v19 but since the fix to the module it's extremely easy and since the change is now 3 years old... I'll close this issue.
